navigator/backend/src/main/scala/com/digitalasset/navigator/UIBackend.scala

```diff
@@ -293,13 +294,14 @@ abstract class UIBackend extends LazyLogging with ApplicationInfoJsonSupport {
 
     val stopAkka = () => Try(Await.result(system.terminate(), 10.seconds))
 
-    if (arguments.startConsole) {
-      console.Console.run(arguments, config, store, graphQL, applicationInfo)
-      // Stop the web server, then the Akka system consuming the ledger API
-      stopServer()
-      partyRefresh.foreach(_.cancel)
-      stopAkka()
-      ()
+    discard {
+      sys.addShutdownHook {
+        // Stop the web server, then the Akka system consuming the ledger API
+        stopServer()
+        partyRefresh.foreach(_.cancel)
+        stopAkka()
+        ()
```

Comment:
This wasn't in a shutdown hook before. Did something change or am I missing something?

Reply:
Nothing changed but before we didn’t do any attempts at clean shutdown outside of Console. Rather than dropping that logic completely, moving it to a shutdown hook seemed more sensible so we use it for the web server.